Sources/AuthFoundation/Utilities/JSONValue.swift

```diff
@@ -0,0 +1,172 @@
+//
+// Copyright (c) 2023-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+
+public enum JSONValueError: Error {
+    case cannotDecode(value: Any)
+}
+
+/// Represent mixed JSON values as instances of `Any`. This is used to expose API response values to Swift native types
+/// where Swift enums are not supported.
+public enum JSONValue: Equatable {
```

Comment:
**Note:** This class came from the `okta-idx-swift` repo, which will be updated to consume AuthFoundation for this feature in another PR.